### PR TITLE
feat: hero block

### LINKED ISSUE #9: Hero Block
Hero block supporting:

- [x] bg image (tbd: use AEM dynamic media)
- [ ] title & c2a
- [ ] video player overlay

Screenshot 2023-08-02 at 16 43 29

### PR DESCRIPTION
**Pull request by mhaack**
Fix #9 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.live/
- After: https://feat-9-header--danaher-ls-aem--hlxsites.hlx.live/

### CODE REVIEW

File: blocks/hero/hero.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    return {
+      id: descriptorParts[0],
+    };
+  }
+  return null;
+};
+
+const toggleVideoOverlay = () => {
+  const modal = document.querySelector(videoModalSelector);
+  if (modal?.classList.contains('hidden')) {
```
Comment (duynguyen) [inline blocks/hero/hero.js:25]: you could also use: `modal?.classList.toggle('hidden')` to replace the if-else.

File: blocks/hero/hero.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const videoModal = div(
+    {
+      class:
+        'video-modal hidden fixed inset-0 bg-black/25 backdrop-brightness-50 flex item-center justify-center overflow-hidden z-50',
+      'aria-modal': 'true',
+      role: 'dialog',
+    },
+    videoContainer,
+  );
+  videoModal.addEventListener('click', toggleVideoOverlay);
```
Comment (duynguyen) [inline blocks/hero/hero.js:67]: just an FYI: won't work in all cases, but another way to add event listener to an element (implicitly by dom-builder):
```
const videoModal = div(
    {
      class:
        'video-modal hidden fixed inset-0 bg-black/25 backdrop-brightness-50 flex item-center justify-center overflow-hidden z-50',
      'aria-modal': 'true',
      role: 'dialog',
      onclick: toggleVideoOverlay,
    },
    videoContainer,
);
```

**Review by duynguyen — APPROVED** [review]
LGTM, just some minor comments

File: blocks/hero/hero.js (hunk 1 above)
Comment (mhaack) [reply blocks/hero/hero.js]: Thanks, fixed.

File: blocks/hero/hero.js (hunk 2 above)
Comment (mhaack) [reply blocks/hero/hero.js]: 👍 updated as well

**Pull request merged**